src/main/java/org/opentripplanner/model/plan/Itinerary.java

```diff
@@ -116,9 +116,11 @@ public class Itinerary {
    */
   public Fare fare = new Fare();
 
-  public Itinerary(List<Leg> legs) {
+  public Itinerary(List<Leg> legs, int generalizedCost) {
     this.legs = List.copyOf(legs);
 
+    this.generalizedCost = generalizedCost;
```

Comment:
Is there ever a case, where this can't be computed from the legs?

Reply:
Well, I'm not sure. It is usually computed from the graph path/or raptor path:

- https://github.com/opentripplanner/OpenTripPlanner/blob/c55951294cb507459ad99a750a1cbc2f49348b28/src/main/java/org/opentripplanner/routing/algorithm/mapping/GraphPathToItineraryMapper.java#L134
- https://github.com/opentripplanner/OpenTripPlanner/blob/0f49f7992ebcfd471562f7ee4e96337c1ccaaa6c/src/main/java/org/opentripplanner/routing/algorithm/mapping/RaptorPathToItineraryMapper.java#L126

Is this equivalent to getting them form the legs?

Reply:
Seems to be:

Reply:
I'm happy to implement that logic if we agree that an itinerary's cost is always the sum of all legs' costs.

This seems to be intuitively correct but maybe there is an edge case that we have not considered. Let's have a quick chat in the dev meeting.